autogpt/core/messaging/base.py

```diff
@@ -1,9 +1,47 @@
-import abc
+from abc import ABC, abstractmethod
+from enum import Enum, StrEnum
+import dataclasses
+from typing import Any, List, Callable, Union
 
+class MessageCategory(Enum):
+    USER_INPUT = 0
+    USER_PROMPT = 1
 
-class Message(abc.ABC):
-    pass
 
+class Role(StrEnum):
+    USER = "user"
+    SYSTEM = "system"
+    ASSISTANT = "assistant"
 
-class MessageBroker(abc.ABC):
-    pass
+
+@dataclasses.dataclass
+class Message(ABC):  # ABC should be within parentheses
+    sender: Role  # Use Any to allow any type of sender
+    message: str
+    kind_of_message: MessageCategory
+
+class MessageBroker(ABC):
```

Comment:
This still needs some discussion and work I think.

Reply:
It also has the fuzziest scope.  We maybe leave this PR til the end of the abstractions since its the most new feature kind of thing we're adding and it'll be easier to think about with the other systems resolved.